src/main/java/frc/robot/DistanceRange.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot;
+
+import java.util.HashMap;
+
+import frc.robot.AutonomousPath.VisionPathName;
+
+public class DistanceRange 
+  implements DistanceRangeInterface{
+
+    HashMap<String, Double> distanceRanges;
+
+    public DistanceRange(){
+        
+        this.distanceRanges = new HashMap<String, Double>();
+
+        distanceRanges.put("PathRedAUpperBound", 10.00);
+        distanceRanges.put("PathRedALowerBound", 8.50);
+        
+        distanceRanges.put("PathBlueAUpperBound", 16.30);
+        distanceRanges.put("PathBlueALowerBound", 16.10);
+
+        distanceRanges.put("PathRedBUpperBound", 10.40);
+        distanceRanges.put("PathRedBLowerBound", 10.30);
+
+        distanceRanges.put("PathBlueBUpperBound", 17.80);
+        distanceRanges.put("PathBlueBLowerBound", 17.50);
+    }
+
+    public VisionPathName getPathForCalculatedDistance(double distanceCalculated){
+
+        AutonomousPath objectAutonomousPath = new AutonomousPath();
+
+        if (distanceCalculated >= getDistanceRanges().get("PathRedALowerBound") && 
```

Comment:
aaaaaaaaaaaaaaaahhhhhhhhhhhhhhh
this should have screamed 'bad' when you were writing it! way too much repetition!

I think you misunderstood what i meant by a distancerange object. I meant a class for which instance has ONE choice. This class, as written, doesnt solve the problem of the huge if/else block at all.

public class DistanceRange { 
    public DistanceRange ( int lowerbound, int upperbound, VisionPathName path ){
           this.lowerbound = lowerbound
           etc
    } 
    private int lowerbound = 0;
    private int upperbound = 0;
     private VisionPathName path;

    public boolean isInRange(int value ){
             if ( value > lowerbound ) && value < upperbound ){
                   return this.path
            }
           else{
                return VisionPathName.UnknownPath
    }
}